src/server/dfly_bench.cc

```diff
@@ -391,12 +417,35 @@ string KeyGenerator::operator()() {
         seq_cursor_ = min_;
       break;
   }
+  string res = prefix_;
+  if (IsClusterEnabled()) {
+    absl::StrAppend(&res, "{", hash_slots_[slot_id], "}");
+  }
+  absl::StrAppend(&res, key_suffix);
+  return res;
+}
 
-  return StrCat(prefix_, key_suffix);
+void KeyGenerator::EnableClusterMode() {
+  hash_slots_.resize(kNumSlots);
+  uint32_t i = 0;
+  uint32_t num_slots_filled = 0;
+
+  // Precompute the hash slots for each of the slot ids so given the slot id
+  // we could generate a key that belongs to that slot.
+  while (num_slots_filled < kNumSlots) {
```

Comment:
I thought we would leave some empty cells but then I saw that we increment the num_slots_filled only when hash slot is non empty so at the end we end up with `precomputed keys` for all hash slots. Noice!